mm2src/coins/eth.rs

```diff
@@ -955,8 +968,11 @@ impl SwapOps for EthCoin {
         Box::new(fut.boxed().compat())
     }
 
-    fn validate_maker_payment(&self, input: ValidatePaymentInput) -> Box<dyn Future<Item = (), Error = String> + Send> {
-        let swap_contract_address = try_fus!(input.swap_contract_address.try_to_address());
+    fn validate_maker_payment(
+        &self,
+        input: ValidatePaymentInput,
+    ) -> Box<dyn Future<Item = (), Error = MmError<ValidatePaymentError>> + Send> {
```

Comment:
I think we can add `type ValidatePaymentFut<T> = Box<dyn Future<Item=T, Error=MmError<ValidatePaymentError>> + Send>`

Reply:
looks good, done